### PR TITLE
Add getUsers call

### PR DESCRIPTION
**Pull request by gneokleo**
- Create a User interface to be used when deserializing JSON

- Use the new User object as part of the return type in getUsers

### CODE REVIEW

File: src/client/resource.ts [hunk 1]
```diff
@@ -16,17 +16,17 @@ abstract class Resource {
   }
 
   /**
-   * Converts a JSON object to a map of string keys to string values.
+   * Converts a JSON object to a map of string keys to V values.
    *
```
Comment (psastras) [inline src/client/resource.ts]: @param V doc

File: test/client/users-test.ts [hunk 2]
```diff
@@ -1,15 +1,57 @@
 import test from 'ava';
 
 import { setupLocalClientMock } from './_utils';
+import Resource from 'src/client/resource';
 import Users from 'src/client/users';
 
 const [underTest, nock] = setupLocalClientMock(Users);
 
 test('get user photos should return a map of users to URLs', async (t) => {
```
Comment (psastras) [inline test/client/users-test.ts]: do we still need this test

File: src/client/users.ts [hunk 3]
```diff
@@ -1,23 +1,34 @@
 import Resource from './resource';
-
+import User from '../model/user';
```
Comment (psastras) [inline src/client/users.ts]: prefer src/model/user

File: test/client/users-test.ts (hunk 2 above)
Comment (gneokleo) [inline test/client/users-test.ts]: this is a different API endpoint

File: src/client/users.ts (hunk 3 above)
Comment (gneokleo) [inline src/client/users.ts]: Done

File: src/client/resource.ts (hunk 1 above)
Comment (gneokleo) [inline src/client/resource.ts]: Done

**Pull request merged**